### DISCUSSION TITLE
Unable to debug uno app on android

### DISCUSSION THREAD

**Discussion by modula2019** [unoplatform/uno#15465] [Q&A] [answered]
Hi support,
Please I am unable to debug my uno app on android whereas I can do the same on windows and web. Debugging only stops for android device/emulator. Deployment works but debugging stops on android device. Windows and web have no problem with debugging and deployment. 
I am using android 13 API 33, .net 8.0, uno 5.1

I will attach the code here for assistance. I have removed the bin and obj folders so as to reduce the zip size.

See the debug log

Start debugging Android application ...
Forwarding debugger port 8880
Detecting existing process
> am start -D -a "android.intent.action.MAIN" -c "android.intent.category.LAUNCHER" -n "com.modula.eDeliveryApp/crc641c0f6372bf9cd76b.MainActivity"
> Starting: Intent { act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] cmp=com.modula.eDeliveryApp/crc641c0f6372bf9cd76b.MainActivity }
[la.eDeliveryApp] Late-enabling -Xcheck:jni
[CompatibilityChangeReporter] Compat change id reported: 171979766; UID 10079; state: ENABLED
[System.out] Sending WAIT chunk
[System.out] Debugger has connected
[System.out] waiting for debugger to settle...
[System.out] waiting for debugger to settle...
[System.out] waiting for debugger to settle...
[System.out] waiting for debugger to settle...
[System.out] waiting for debugger to settle...
[System.out] waiting for debugger to settle...
[System.out] waiting for debugger to settle...
[System.out] waiting for debugger to settle...
[System.out] debugger detached?
[ziparchive] Unable to open '/data/app/~~HvqzGfo5N4g7zjjfGIGAsg==/com.modula.eDeliveryApp-wow_XeEB0oR18Ynb7U6C1g==/base.dm': No such file or directory
[ziparchive] Unable to open '/data/app/~~HvqzGfo5N4g7zjjfGIGAsg==/com.modula.eDeliveryApp-wow_XeEB0oR18Ynb7U6C1g==/base.dm': No such file or directory
[GraphicsEnvironment] ANGLE Developer option for 'com.modula.eDeliveryApp' set to: 'default'
[GraphicsEnvironment] ANGLE GameManagerService for com.modula.eDeliveryApp: false
[GraphicsEnvironment] Neither updatable production driver nor prerelease driver is supported.
[NetworkSecurityConfig] No Network Security Config specified, using platform default
[NetworkSecurityConfig] No Network Security Config specified, using platform default
[debug-app-helper] Checking if libmonodroid was unpacked to /data/app/~~HvqzGfo5N4g7zjjfGIGAsg==/com.modula.eDeliveryApp-wow_XeEB0oR18Ynb7U6C1g==/lib/x86_64/libmonodroid.so
[debug-app-helper] Native libs extracted to /data/app/~~HvqzGfo5N4g7zjjfGIGAsg==/com.modula.eDeliveryApp-wow_XeEB0oR18Ynb7U6C1g==/lib/x86_64, assuming application/android:extractNativeLibs == true
[debug-app-helper] Setting up for DSO lookup in app data directories
[debug-app-helper] Added filesystem DSO lookup location: /data/app/~~HvqzGfo5N4g7zjjfGIGAsg==/com.modula.eDeliveryApp-wow_XeEB0oR18Ynb7U6C1g==/lib/x86_64
[debug-app-helper] Using runtime path: /data/app/~~HvqzGfo5N4g7zjjfGIGAsg==/com.modula.eDeliveryApp-wow_XeEB0oR18Ynb7U6C1g==/lib/x86_64
[debug-app-helper] checking directory: `/data/user/0/com.modula.eDeliveryApp/files/.__override__/lib`
[debug-app-helper] directory does not exist: `/data/user/0/com.modula.eDeliveryApp/files/.__override__/lib`
[debug-app-helper] Checking whether Mono runtime exists at: /data/user/0/com.modula.eDeliveryApp/files/.__override__/libmonosgen-2.0.so
[debug-app-helper] Checking whether Mono runtime exists at: /data/app/~~HvqzGfo5N4g7zjjfGIGAsg==/com.modula.eDeliveryApp-wow_XeEB0oR18Ynb7U6C1g==/lib/x86_64/libmonosgen-2.0.so
[debug-app-helper] Mono runtime found at: /data/app/~~HvqzGfo5N4g7zjjfGIGAsg==/com.modula.eDeliveryApp-wow_XeEB0oR18Ynb7U6C1g==/lib/x86_64/libmonosgen-2.0.so
[la.eDeliveryApp] Attempt to remove non-JNI local reference, dumping thread
[DOTNET] JNI_OnLoad: JNI_OnLoad in pal_jni.c
[DOTNET] GetOptionalClassGRef: optional class com/android/org/conscrypt/OpenSSLEngineImpl was not found
[monodroid] Creating public update directory: `/data/user/0/com.modula.eDeliveryApp/files/.__override__`
[la.eDeliveryApp] Attempt to remove non-JNI local reference, dumping thread
[monodroid-debug] Trying to initialize the debugger with options: --debugger-agent=transport=dt_socket,loglevel=0,address=127.0.0.1:8880,server=y,embedding=1,timeout=1707874042
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/System.Private.CoreLib.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly eDeliveryApp.Mobile.dll
[monodroid-gc] GREF GC Threshold: 46080
[monodroid-assembly] open_from_bundles: failed to load assembly Mono.Android.dll
[monodroid-assembly] open_from_bundles: failed to load assembly System.Runtime.dll
[monodroid-assembly] open_from_bundles: failed to load assembly Java.Interop.dll
[monodroid-assembly] open_from_bundles: failed to load assembly System.Runtime.InteropServices.dll
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/eDeliveryApp.Mobile.dll
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/Mono.Android.dll [External]
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/System.Runtime.dll [External]
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/Java.Interop.dll [External]
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/System.Runtime.InteropServices.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly System.Collections.dll
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/System.Collections.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly Mono.Android.Runtime.dll
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/Mono.Android.Runtime.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly System.Threading.dll
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/System.Threading.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly System.Threading.Thread.dll
[monodroid-assembly] open_from_bundles: failed to load assembly System.Diagnostics.StackTrace.dll
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/System.Threading.Thread.dll [External]
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/System.Diagnostics.StackTrace.dll [External]
[la.eDeliveryApp] Attempt to remove non-JNI local reference, dumping thread
[monodroid-assembly] open_from_bundles: failed to load assembly Uno.UI.dll
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/Uno.UI.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly System.Memory.dll
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/System.Memory.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly System.Reflection.Emit.Lightweight.dll
[monodroid-assembly] open_from_bundles: failed to load assembly System.Reflection.Emit.ILGeneration.dll
[monodroid-assembly] open_from_bundles: failed to load assembly System.Reflection.Primitives.dll
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/System.Reflection.Emit.Lightweight.dll [External]
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/System.Reflection.Emit.ILGeneration.dll [External]
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/System.Reflection.Primitives.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly System.Linq.dll
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/System.Linq.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly Mono.Android.Export.dll
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/Mono.Android.Export.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly System.Xml.ReaderWriter.dll
[monodroid-assembly] open_from_bundles: failed to load assembly System.Private.Xml.dll
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/System.Xml.ReaderWriter.dll [External]
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/System.Private.Xml.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly System.Reflection.Emit.dll
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/System.Reflection.Emit.dll [External]
Loaded assembly: __callback_factory__ [External]
[monodroid-assembly] open_from_bundles: failed to load assembly System.Collections.NonGeneric.dll
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/System.Collections.NonGeneric.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly Uno.Extensions.Reactive.dll
[monodroid-assembly] open_from_bundles: failed to load assembly netstandard.dll
[monodroid-assembly] open_from_bundles: failed to load assembly Microsoft.Extensions.Logging.Abstractions.dll
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/Uno.Extensions.Reactive.dll [External]
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/netstandard.dll [External]
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/Microsoft.Extensions.Logging.Abstractions.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly System.Console.dll
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/System.Console.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly Uno.Core.Extensions.Logging.Singleton.dll
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/Uno.Core.Extensions.Logging.Singleton.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly CommonServiceLocator.dll
[monodroid-assembly] open_from_bundles: failed to load assembly System.ComponentModel.dll
[monodroid-assembly] open_from_bundles: failed to load assembly Microsoft.Extensions.Logging.dll
[monodroid-assembly] open_from_bundles: failed to load assembly System.Collections.Concurrent.dll
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/CommonServiceLocator.dll [External]
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/System.ComponentModel.dll [External]
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/Microsoft.Extensions.Logging.dll [External]
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/System.Collections.Concurrent.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly Microsoft.Extensions.Options.dll
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/Microsoft.Extensions.Options.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly Uno.Extensions.Reactive.UI.dll
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/Uno.Extensions.Reactive.UI.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly Uno.Extensions.Core.dll
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/Uno.Extensions.Core.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly System.Collections.Immutable.dll
[monodroid-assembly] open_from_bundles: failed to load assembly Microsoft.Bcl.AsyncInterfaces.dll
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/System.Collections.Immutable.dll [External]
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/Microsoft.Bcl.AsyncInterfaces.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly Uno.Diagnostics.Eventing.dll
[monodroid-assembly] open_from_bundles: failed to load assembly Uno.Foundation.dll
[monodroid-assembly] open_from_bundles: failed to load assembly Uno.Foundation.Logging.dll
[monodroid-assembly] open_from_bundles: failed to load assembly Uno.dll
[monodroid-assembly] open_from_bundles: failed to load assembly Uno.UI.Composition.dll
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/Uno.Diagnostics.Eventing.dll [External]
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/Uno.Foundation.dll [External]
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/Uno.Foundation.Logging.dll [External]
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/Uno.dll [External]
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/Uno.UI.Composition.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly eDeliveryApp.dll
[monodroid-assembly] open_from_bundles: failed to load assembly Microsoft.Extensions.Hosting.Abstractions.dll
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/eDeliveryApp.dll
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/Microsoft.Extensions.Hosting.Abstractions.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly Uno.UI.Dispatching.dll
[monodroid-assembly] open_from_bundles: failed to load assembly Uno.UI.BindingHelper.Android.dll
[monodroid-assembly] open_from_bundles: failed to load assembly System.Numerics.Vectors.dll
[monodroid-assembly] open_from_bundles: failed to load assembly System.Private.Uri.dll
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/Uno.UI.Dispatching.dll [External]
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/Uno.UI.BindingHelper.Android.dll [External]
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/System.Numerics.Vectors.dll [External]
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/System.Private.Uri.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly System.Text.RegularExpressions.dll
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/System.Text.RegularExpressions.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly System.ObjectModel.dll
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/System.ObjectModel.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly System.Runtime.Loader.dll
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/System.Runtime.Loader.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly Uno.UI.RemoteControl.dll
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/Uno.UI.RemoteControl.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly Uno.UI.RemoteControl.Messaging.dll
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/Uno.UI.RemoteControl.Messaging.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly System.Net.WebSockets.dll
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/System.Net.WebSockets.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly System.Net.WebSockets.Client.dll
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/System.Net.WebSockets.Client.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly System.Net.Http.dll
[monodroid-assembly] open_from_bundles: failed to load assembly System.Net.Primitives.dll
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/System.Net.Http.dll [External]
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/System.Net.Primitives.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly System.Net.Security.dll
[monodroid-assembly] open_from_bundles: failed to load assembly System.Security.Cryptography.dll
[monodroid-assembly] open_from_bundles: failed to load assembly System.Net.WebHeaderCollection.dll
[monodroid-assembly] open_from_bundles: failed to load assembly System.Collections.Specialized.dll
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/System.Net.Security.dll [External]
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/System.Security.Cryptography.dll [External]
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/System.Net.WebHeaderCollection.dll [External]
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/System.Collections.Specialized.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly Microsoft.Win32.Primitives.dll
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/Microsoft.Win32.Primitives.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly System.Diagnostics.DiagnosticSource.dll
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/System.Diagnostics.DiagnosticSource.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly System.Diagnostics.Tracing.dll
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/System.Diagnostics.Tracing.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly System.Net.Quic.dll
[monodroid-assembly] open_from_bundles: failed to load assembly System.Threading.Channels.dll
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/System.Net.Quic.dll [External]
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/System.Threading.Channels.dll [External]
Thread started: <Thread Pool> #2
Thread started: .NET TP Gate #3
Thread started: <Thread Pool> #4
[monodroid-assembly] open_from_bundles: failed to load assembly System.Net.Sockets.dll
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/System.Net.Sockets.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly System.Runtime.Intrinsics.dll
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/System.Runtime.Intrinsics.dll [External]
Thread started: <Thread Pool> #5
[monodroid-assembly] open_from_bundles: failed to load assembly System.Threading.ThreadPool.dll
Loaded assembly: /data/data/com.modula.eDeliveryApp/files/.__override__/System.Threading.ThreadPool.dll [External]
[la.eDeliveryApp] * Assertion at /__w/1/s/src/mono/mono/metadata/object.c:657, condition `lock->done' not met
[libc] Fatal signal 6 (SIGABRT), code -1 (SI_QUEUE) in tid 11681 (.NET TP Worker), pid 11597 (la.eDeliveryApp)
Thread started: <Thread Pool> #6
[la.eDeliveryApp] * Assertion at /__w/1/s/src/mono/mono/metadata/object.c:657, condition `lock->done' not met

Thanks in advance

[eDeliveryApp.zip](https://github.com/unoplatform/uno/files/14274353/eDeliveryApp.zip)

**Comment by jeromelaban** [comment, marked as answer]
Thanks for the report, please see https://github.com/unoplatform/uno/discussions/15044 for a resolution.

**Reply by modula2019** [reply]
Thanks so much. This solves the problem
